src/destination/mock.ts

```diff
@@ -0,0 +1,124 @@
+import fastRedact from 'fast-redact';
+
+export type MockOptions = {
+  /**
+   * Properties to replace with a static `-` before recording the logging call.
+   *
+   * List non-determistic properties like `latency` to stabilise snapshots.
+   *
+   * See `fast-redact` for supported syntax:
+   * https://github.com/davidmarkclements/fast-redact/blob/v3.5.0/readme.md#paths--array
+   */
+  redact?: string[];
+
+  /**
+   * Properties to remove before recording the logging call.
+   *
+   * List common properties like `timestamp` to declutter test assertions.
+   *
+   * See `fast-redact` for supported syntax:
+   * https://github.com/davidmarkclements/fast-redact/blob/v3.5.0/readme.md#paths--array
+   */
+  remove?: string[];
+};
+
+export const DEFAULT_MOCK_OPTIONS = Object.freeze({
+  redact: [
+    'headers["host"]',
+    'headers["x-request-id"]',
+    'latency',
+    '["x-request-id"]',
+  ],
+
+  remove: ['environment', 'name', 'timestamp', 'version'],
+} as const satisfies MockOptions);
+
+export const createStdoutMock = (opts: MockOptions) => {
+  const redact = fastRedact({
+    censor: '-',
+    paths: opts.redact ?? DEFAULT_MOCK_OPTIONS.redact,
+    serialize: false,
+    remove: false,
+    strict: true,
+  });
+
+  const remove = fastRedact({
+    censor: undefined,
+    paths: opts.remove ?? DEFAULT_MOCK_OPTIONS.remove,
+    serialize: JSON.stringify,
+    remove: true,
+    strict: true,
+  });
+
+  const calls: object[] = [];
+
+  return {
+    /**
+     * The list of logging calls recorded to date.
+     *
+     * This may be asserted against in a test case.
+     *
+     * ```typescript
+     * expect(stdoutMock.calls).toMatchSnapshot();
+     * ```
+     */
+    calls,
+
+    /**
+     * Convenience method to clear the logging calls recorded to date.
+     *
+     * This may be hooked up to a test runner lifecycle event.
+     *
+     * ```typescript
+     * afterEach(stdoutMock.clear);
+     * ```
+     */
+    clear: (): void => {
+      calls.length = 0;
+    },
+
+    /**
+     * Convenience method to retrieve a solitary logging call.
+     *
+     * Throws an error if more or less than 1 call has been recorded.
+     *
+     * This may be asserted against in a test case.
+     *
+     * ```typescript
+     * expect(stdoutMock.onlyCall()).toMatchSnapshot();
+     * ```
+     */
+    onlyCall: (): object => {
```

Comment:
Can we think of any other convenience methods worth implementing in a first cut?